sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -382,14 +381,16 @@ public FormRecognizerServiceVersion getServiceVersion() {
 
     private Function<PollingContext<OperationResult>, Mono<OperationResult>> receiptStreamActivationOperation(
         Flux<ByteBuffer> data, long length, FormContentType formContentType, boolean includeTextDetails) {
-        return (pollingContext) -> {
+        Objects.requireNonNull(data, "'data' is required and cannot be null.");
+
+        return pollingContext -> {
             try {
-                Objects.requireNonNull(data, "'data' is required and cannot be null.");
-                Objects.requireNonNull(formContentType, "'formContentType' is required and cannot be null.");
-                return service.analyzeReceiptAsyncWithResponseAsync(ContentType.fromString(formContentType.toString()),
-                    data, length, includeTextDetails)
-                    .map(response -> new OperationResult(
-                        parseModelId(response.getDeserializedHeaders().getOperationLocation())));
+                return data.next()
```

Comment:
Thanks, Shawn, for simplifying this.

This looks better, but it assumes that the first ByteBuffer from data is at least 2 or 4 bytes, which may not always the case. For example, if the data arrives in following structure, then next() operator is not sufficient:

```java
Flux<ByteBuffer> data = Flux.fromArray(new ByteBuffer [] {
    ByteBuffer.wrap(new byte[]{(byte) 0xff}),
    ByteBuffer.wrap(new byte[]{(byte) 0xd8}),
    ByteBuffer.wrap(new byte[]{3}),
    ByteBuffer.wrap(new byte[]{4})
});
```

may be something like below which looks for header even if the header bytes are split across chunks, (please test it, also return type should be ContentType)

```java
private static Mono<String> detectContentType(Flux<ByteBuffer> data) {
    byte[] header = new byte[4];
    int[] written = new int[] {0};
    String[] contentType = new String[] {null};
    return data.map(chunk -> {
        final int len = chunk.remaining();
        for (int i = 0 ;i < len; i++) {
            header[written[0]] = chunk.get(i);
            written[0]++;
            if (written[0] == 2) {
                if (isJpeg(header)) {
                    contentType[0] = "IMAGE_JPEG";
                    // Got a matching two bytes, return false so that takeWhile
                    // can cut the subscription on data
                    return false;
                } else {
                    // two bytes didn't match, continue to read next byte
                    continue;
                }
            }
            if (written[0] == 4) {
                if (isPdf(header)) {
                    contentType[0] = "APPLICATION_PDF";
                } else if (isPng(header)) {
                    contentType[0] = "IMAGE_PNG";
                } else if (isTiff(header)) {
                    contentType[0] = "IMAGE_TIFF";
                }
                // Got a four bytes matching or not, either way no need to read more byte return false 
               // so that takeWhile can cut the subscription on data
                return false;
            }
        }
        // current chunk don't have enough bytes so return true to get next Chunk if there is one.
        return true;
    })
    .takeWhile(doContinue -> doContinue == true)
    .then(Mono.defer(() -> {
        if (contentType[0] != null) {
            return Mono.just(contentType[0]);
        } else {
            return Mono.error(new RuntimeException("Content type could not be detected. " +
                "Should use other overload API that takes content type."));
        }
    }));
}

private static boolean isJpeg(byte[] header) {
    return (header[0] == (byte) 0xff && header[1] == (byte) 0xd8);
}

private static boolean isPdf(byte[] header) {
    return header[0] == (byte) 0x25
        && header[1] == (byte) 0x50
        && header[2] == (byte) 0x44
        && header[3] == (byte) 0x46;
}

private static boolean isPng(byte[] header) {
   return header[0] == (byte) 0x89
        &&  header[1] == (byte) 0x50
        && header[2] == (byte) 0x4e
        && header[3] == (byte) 0x47;
}

private static boolean isTiff(byte[] header) {
    return (header[0] == (byte) 0x49
        && header[1] == (byte) 0x49
        && header[2] == (byte) 0x2a
        && header[3] == (byte) 0x0)
        // big-endian
        || (header[0] == (byte) 0x4d
        && header[1] == (byte) 0x4d
        && header[2] == (byte) 0x0
        && header[3] == (byte) 0x2a);
}
```

Reply:
also, it may be more clean/readable to have this scanning only if user didn't provide the contentType, so that we can avoid double subscription if not needed.

```java
if (contentType !=) {
   return callAPI(data, contentType);
} else {
  return detectContentType(data)
     .flatmap(contentType -> {
         return callAPI(data, contentType);
     }); 
}
```

Reply:
We can further simplify the JPEG section as a valid JPEG contains a minimum of 4 bytes: https://en.wikipedia.org/wiki/JPEG#Syntax_and_structure

```java
for (int i = 0; i < len; i++) {
	header[written[0]] = chunk.get(i);
	written[0]++;
	if (written[0] == 4) {
		if (isJpeg(header)) {
			contentType[0] = "IMAGE_JPEG";
		} else if (isPdf(header)) {
			contentType[0] = "APPLICATION_PDF";
		} else if (isPng(header)) {
			contentType[0] = "IMAGE_PNG";
		} else if (isTiff(header)) {
			contentType[0] = "IMAGE_TIFF";
		}
		// Got a four bytes matching or not, either way no need to read more byte return false 
	   // so that takeWhile can cut the subscription on data
		return false;
	}
}      
```

Reply:
@srnagar interesting, I just copied from the existing code already in this PR. I wonder the two byte check was agreed across all language sdks, if so we should communicate correct look up (as you suggested) to others.

Reply:
To find the content type, the first 2 bytes (Start of image) are sufficient. However, a valid jpeg will also have 2 trailing bytes (End of image). So, reading 4 bytes before determining the content type is okay and doesn't need special logic to stop after reading 2 bytes.

Reply:
synced with @srnagar, @mssfang -  so JPEG has 2 leading bytes that mark the start of image and 2 trailing bytes that mark the end of image, we can read 4 bytes but have to check only the first two bytes, which simplifies the code like Srikanta pointed.